packages/rs-drive-abci/src/platform_types/snapshot/mod.rs

```diff
@@ -0,0 +1,169 @@
+    pub hash: [u8; 32],
+    /// Metadata
+    pub metadata: Vec<u8>,
+}
+
+/// Snapshot manager is responsible for creating and managing snapshots to keep only the certain
+/// number of snapshots and remove the old ones
+pub struct Manager {
+    freq: i64,
+    number_stored_snapshots: usize,
+    checkpoints_path: String,
+}
+
+impl Manager {
+    /// Create a new instance of snapshot manager
+    pub fn new(db_path: String, number_stored_snapshots: Option<usize>, freq: Option<i64>) -> Self {
+        let mut manager = Self {
+            freq: DEFAULT_FREQ,
+            number_stored_snapshots: DEFAULT_NUMBER_OF_SNAPSHOTS,
+            checkpoints_path: db_path,
+        };
+        if freq.is_some() {
+            manager.freq = freq.unwrap()
+        }
+        if number_stored_snapshots.is_some() {
+            manager.number_stored_snapshots = number_stored_snapshots.unwrap()
+        }
+        manager
+    }
+
+    /// Return a persisted list of snapshots
+    pub fn get_snapshots(&self, grove: &GroveDb) -> Result<Vec<Snapshot>, Error> {
+        let data = grove
+            .get_aux(SNAPSHOT_KEY, None)
+            .unwrap()
+            .map_err(|e| Error::Drive(GroveDB(e)))?;
+
+        match data {
+            Some(data) => {
+                let conf = config::standard();
+                let (mut decoded, _): (Vec<Snapshot>, usize) =
+                    bincode::decode_from_slice(data.as_slice(), conf)
+                        .map_err(|e| Error::Drive(Drive(DriveError::Snapshot(e.to_string()))))?;
+                decoded.sort_by(|a, b| a.height.cmp(&b.height));
+                Ok(decoded)
+            }
+            None => Ok(vec![]),
+        }
+    }
+
+    /// Create a new snapshot for the given height, if a height is not a multiple of N,
+    /// it will be skipped.
+    pub fn create_snapshot(&self, grove: &GroveDb, height: i64) -> Result<(), Error> {
+        if height == 0 || height % self.freq != 0 {
+            return Ok(());
+        }
+        let mut checkpoint_path = self.checkpoints_path.to_owned();
+        checkpoint_path.push_str("/");
+        checkpoint_path.push_str(height.to_string().as_str());
+        grove
+            .create_checkpoint(checkpoint_path.to_string())
+            .map_err(|e| Error::Drive(GroveDB(e)))?;
+
+        let root_hash = grove
+            .root_hash(None)
+            .unwrap()
+            .map_err(|e| Error::Drive(Drive(DriveError::Snapshot(e.to_string()))))?;
+
+        let snapshot = Snapshot {
+            height,
+            version: 0,
+            path: checkpoint_path,
+            hash: root_hash as [u8; 32],
+            metadata: vec![],
+        };
+
+        let mut snapshots = self.get_snapshots(grove)?;
+        snapshots.push(snapshot);
+        snapshots = self.prune_snapshots(snapshots)?;
+        self.save_snapshots(grove, snapshots)
+    }
+
+    fn prune_snapshots(&self, snapshots: Vec<Snapshot>) -> Result<Vec<Snapshot>, Error> {
```

Comment:
All good. The only tiny thing I would do differently is rename this method because when I first saw this I was like "why do we prune all snapshots and then save them again?". I probably would name it as "prune_excess_snapshots"